src/main.rs

```diff
@@ -272,6 +374,23 @@ fn release_build(musl: bool) -> Result<(), Error> {
     Ok(())
 }
 
+/// Build a source tarball from the current (project) directory.
+/// Utilises `cargo --publish --dry-run` under the hood to do the packaging.
+fn source_tarball(cargo_target: &Path, output: &Path, config: &Config) -> Result<(), Error> {
+    let args = ["publish", "--dry-run", "--allow-dirty"];
+    let status = Command::new("cargo").args(args).status()?;
```

Comment:
What if, instead of building the tarball ourselves, we pull one automatically based on a release tag? That's the strategy I use for [Aura's main package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=aura) on the AUR.

Reply:
That should be OK, this will require author to publish to github using the exact tag format we are expecting so that we can then pull it to generate the checksums. Are you OK with that? I was worried that it was kind of an implicit behaviour that would need extra documentation, particularly since it deviates from the default. Dare I say it but I feel like this would have been a use case for source=url

Reply:
> the exact tag format we are expecting

Isn't the tag format uniform, at least with Github? Always something like `v1.2.3`?

Reply:
Github tags are free text entry - here's an example of one I made up on a spare repository.
https://github.com/nick42d/ratatuirepro/releases/tag/version-1.501.22